### PR TITLE
implements a pretty printer

### PR DESCRIPTION
**Pull request by KFlash**
This replace the current Kataw printer and implements a pretty printer like Prettier that prints Kataw AST and types.

**TODO**

- [x] Finish style formatting

- [ ] Finish options

- [ ] Implements same comment attach algorithm as in current printer

- [ ] performance

### CODE REVIEW

**Comment by KFlash** [comment]
@aladdin-add Can you play around with this and let me know where I need to do some improvements?

**Comment by KFlash** [comment]
@aladdin-add Can you play around with this and let me know where I need to do some improvements? It's important because everything need to be "correct" before I start to optimize it. Fixing performance can only be done if I use bitwise masks.

**Comment by aladdin-add** [comment]
can you update the docs for the available options?  so, I can make a comparison, and add a section like "migrate from prettier".

**Comment by KFlash** [comment]
@aladdin-add Don't expect the format to be 1:1 to Prettier, because this isn't Prettier :)  The style format is similar, but I improved the readability a few places.  For example how conditional expression are printed are different and binaries. 

Prettier also got some nasty bugs where long lines didn't break - for example in obj literal.  I solved this in Kataw and added an alternative and more readable style format for that.

**Comment by KFlash** [comment]
I will add the options in the readme in the printer folder.

**Comment by KFlash** [comment]
@aladdin-add I updated the options info in the readme and added a few new options known from ESLint

**Comment by KFlash** [comment]
@aladdin-add I implemented the 'computedPropertySpacing' option from ESLint and updated the readme. I think it's enough options now?

**Comment by KFlash** [comment]
@aladdin-add can you try to style union types?

File: test/__snapshot__/ecma262/miscellaneous/should-pass/gen/stand-alone/x002fx002a_x002ax002f_x002fx002a_optional_SingleLineDelimitedCommentSequence_x002ax002f--x003ethe_comment_extends_to_these_chara.md [hunk 1]
```diff
@@ -39,6 +39,9 @@
 
 ```javascript
 
+/*
+  */
+/* optional SingleLineDelimitedCommentSequence */
 ```
```
Comment (KFlash) [inline test/__snapshot__/ecma262/miscellaneous/should-pass/gen/stand-alone/x002fx002a_x002ax002f_x002fx002a_optional_SingleLineDelimitedCommentSequence_x002ax002f--x003ethe_comment_extends_to_these_chara.md]: @aladdin-add Look at this test. It's an 1:1 of original source. This is what will happen when we track whitespace in the correct way.

**Comment by KFlash** [comment]
@aladdin-add I'm closing this. I can't see any reason why I should spend more time finish this. If there are any interest for such a printer I can re-open. For now Kataw has a printer.

**Comment by KFlash** [comment]
@aladdin-add this pr suddenly got conflicts and I just merge to master now. There are no point to have 2 printers.

**Pull request closed without merging**